test/integration/image-optimizer/test/index.test.js

```diff
@@ -0,0 +1,282 @@
+    const opts = { headers: { accept: 'image/png' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/png')
+  })
+
+  it('should resize relative url with invalid accept header as png', async () => {
+    const query = { url: '/test.png', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/invalid' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/png')
+  })
+
+  it('should resize relative url with invalid accept header as gif', async () => {
+    const query = { url: '/test.gif', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/invalid' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/gif')
+  })
+
+  it('should resize relative url with invalid accept header as svg', async () => {
+    const query = { url: '/test.svg', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/invalid' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/svg+xml')
+  })
+
+  it('should resize relative url with invalid accept header as tiff', async () => {
+    const query = { url: '/test.tiff', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/invalid' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/tiff')
+  })
+
+  it('should resize relative url and wildcard accept header as webp', async () => {
+    const query = { url: '/test.png', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/*' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/webp')
+  })
+
+  it('should resize absolute url from localhost', async () => {
+    const url = `http://localhost:${appPort}/test.png`
+    const query = { url, w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/webp' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/webp')
+  })
+
+  it('should use cached image file when parameters are the same', async () => {
+    const query = { url: '/test.png', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/webp' } }
+
+    const res1 = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res1.status).toBe(200)
+    expect(res1.headers.get('Content-Type')).toBe('image/webp')
+    const json1 = await fsToJson(imagesDir)
+    expect(json1).toBeTruthy()
```

Comment:
This should be truthy unless the dir doesn't exist which will cause it to throw an error right? Should we check for the number of output items we expect instead?

Reply:
This is more of a sanity check to make sure the `fsToJson()` was working and we don't accidentally compare null to null below on line 218.

The reason why this isn't checking the exact FS contents is because it changes depending on how many tests run before this one.

For some reason, why I tried to delete the cache contents at the begging of this test, it would fail so I instead delete the cache in the `afterAll` call below.

I'll see if this is still a problem 🤔

Reply:
I pushed an update to this test so its more accurate now, thanks 👍